sig/datadog/core/remote/client.rbs

```diff
@@ -0,0 +1,50 @@
+module Datadog
+  module Core
+    module Remote
+      class Client
+        attr_reader transport: Datadog::Core::Transport::Config::Transport
+
+        attr_reader repository: Configuration::Repository
+
+        attr_reader id: ::String
+
+        def initialize: (Datadog::Core::Transport::Config::Transport transport, ?repository: Configuration::Repository) -> void
+
+        def sync: () -> void
+
+        class SyncError < StandardError
+        end
+
+        private
+
+        def payload: () ->  ::Hash[Symbol, untyped]
```

Comment:
I guess we could improve this `untyped` given that we expect only a specific set of types here.

Reply:
Sounds good. We can tackle that on a follow up PR